nix/lib/purescript.nix

```diff
@@ -1,129 +1,52 @@
-  ];
-
-  packagesJson = "${src}/packages.json";
-
-  cleanSrcs = lib.cleanSourceWith {
-    filter = lib.cleanSourceFilter;
-    src = lib.cleanSourceWith {
-      filter = (path: type: !(lib.elem (baseNameOf path)
-        [ ".spago" ".spago2nix" "generated" "generated-docs" "output" "dist" "node_modules" ".psci_modules" ".vscode" ]));
-      inherit src;
-    };
-  };
-
-  purescriptSources = [
-    "src/**/*.purs"
-    "test/**/*.purs"
-    "generated/**/*.purs"
-    ".spago/*/*/src/**/*.purs"
-  ] ++ additionalPurescriptSources;
+  # Cleans the source based on the patterns in ./.gitignore and the additionalIgnores
+  cleanSrcs = nix-gitignore.gitignoreSource [ "/*.adoc" "/*.nix" ] src;
 
+  # A store path containing [ node_modules/ package.json  package-lock.json ]
+  nodeModules = buildNodeModules { projectDir = src; buildInputs = additionalNpmBuildInputs; };
 in
-yarn2nix-moretea.mkYarnPackage {
-  inherit name packageJSON yarnLock yarnNix checkPhase passthru nodejs;
+stdenv.mkDerivation {
+  inherit name;
   src = cleanSrcs;
-
-  pkgConfig = {
-    "libxmljs" = {
-      buildInputs = [ nodejs python2 ];
-      postInstall = ''
-        # To deal with some OSX setups we need to use the version of node-gyp that's patched in
-        # https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/web/nodejs/nodejs.nix#L106
-        ${nodejs}/lib/node_modules/npm/bin/node-gyp-bin/node-gyp --tarball ${nodejs-headers} rebuild
-      '';
-    };
-  };
-
-  buildInputs = [ cacert ];
-
-  nativeBuildInputs = [ git easyPS.purs easyPS.spago easyPS.psc-package nodePackages.node-gyp nodejs python2 ];
-
+  buildInputs = [ nodeModules easyPS.purs easyPS.spago easyPS.psc-package ];
   buildPhase = ''
     export HOME=$NIX_BUILD_TOP
-    # node-sass is terrible and we need to fix the binaries otherwise it will try and download them
-    export SASS_BINARY_DIR=${nodeSassBinDir}
-
-    # Ensure that the shell expands 'foo/**/*.purs' to include 'foo/*.purs'.
     shopt -s globstar
-
-    # Recent versions of yarn2nix moved the package source
-    # into deps/${name} with a .yarnrc that magically uses
-    # that as the PWD. This doesn't work for us since we're
-    # not just using yarn, so we undo that.
-    # (The node_mdules is just a symlink to something empty,
-    # the real one seems to be in the root, I don't understand
-    # why. We remove it so we can copy without conflicts.)
-    rm deps/${name}/node_modules
-    cp -R deps/${name}/* .
-    rm .yarnrc
-
-    # Put links to the generated and common source in the correct place.
+    ln -s ${nodeModules}/node_modules node_modules
     ln -s ${psSrc} generated
     ln -s ${webCommon} ../web-common
 
-    # Ask spago to make the PureScript packages available.
     sh ${spagoPackages.installSpagoStyle}
-
-    # Compile everything.
-    purs compile ${builtins.concatStringsSep " " purescriptSources}
-
-    # Build the frontend.
-    yarn --offline webpack
+    sh ${spagoPackages.buildSpagoStyle}
```

Comment:
This just runs `spago` for us like before - just that we don't have to construct the command ourselves.